### ISSUE TITLE
Question - Injecting activityFragmentManager in a different way

### ISSUE THREAD

**Issue by kdani41** [vestrel00/android-dagger-butterknife-mvp#54]
Thanks for crunching all these important libraries into one solid sample. Really appreciate it :)

My question might be slight off from the library but i'm using the same approach. 
I have a small scenario where i have this following structure but for some reason I'm not in luck.

```java
@Singleton
@Component(modules = { AppModule.class,
        ActivityModule.class,
        AndroidSupportInjectionModule.class })
public interface AppComponent extends AndroidInjector<App> {

    @Override
    void inject(App application);

    @Component.Builder interface Builder {

        @BindsInstance
        AppComponent.Builder application(App application);

        AppComponent build();
    }
}
```

ActivityModule.class

```java
    @PerActivity
    @ContributesAndroidInjector(modules = BaseActivityModule.class)
    abstract BaseActivity baseActivity();
```

BaseActivityModule.class
```java
    static final String ACTIVITY_FRAGMENT_MANAGER = "ACTIVITY_FRAGMENT_MANAGER";

    @PerActivity
    @Named(ACTIVITY_FRAGMENT_MANAGER)
    @Provides
    static FragmentManager activityFragmentManager(BaseActivity activity) {
        return activity.getSupportFragmentManager();
    }
```

BaseAcitivity.class
```java
public abstract class BaseActivity extends DaggerAppCompatActivity {

    @Named(ACTIVITY_FRAGMENT_MANAGER)
    @Inject
    FragmentManager fragmentManager;
}
```

So even though I'm providing my fragment manager in BaseActivityModule.class dagger is throwing this following compilation error. I even tried with just Activity instead of BaseActivity as my input parameter in BaseActivityModule. Even then I land up in this same issue. Not sure what exactly I'm screwing up. So any help is appreciated. Thanks in advance :)

_2nd Attempt_
I also tried using includes in my child activity modules (ABCActivityModule.class) 

`@Module(includes = BaseActivityModule.class)` 

without using 
`@ContributesAndroidInjector(modules = BaseActivityModule.class)` inside `AcitivityModule.class`

_Result_ - results me to same error
```java
Error:(17, 8) error: [dagger.android.AndroidInjector.inject(T)] @javax.inject.Named("ACTIVITY_FRAGMENT_MANAGER") android.support.v4.app.FragmentManager cannot be provided without an @Provides- or @Produces-annotated method.
@javax.inject.Named("ACTIVITY_FRAGMENT_MANAGER") android.support.v4.app.FragmentManager is injected at
com.abc.views.base.BaseActivity.fragmentManager
com.abc.views.def.ABCActivity is injected at
dagger.android.AndroidInjector.inject(arg0)
A binding with matching key exists in component: om.abc.views.base.BaseActivity_BaseActivity.BaseActivitySubcomponent
```

**Comment by vestrel00** [comment]
@kdani41, I tried a couple of things to reproduce the issue on my end with no luck. I might be missing a few details. Could you trim your entire application to just the following classes with just the bare minimum code (just enough to inject that `FragmentManager`) to reproduce the issue;

1. App
2. AppComponent
3. AppModule
4. ActivityModule
5. BaseActivity
6. BaseActivityModule
7. ABCActivity
8. ABCActivityComponent
9. ABCActivityModule

Then, either copy and paste all Java code as a comment here or create a gist for it so I can take a look there.

**Comment by kdani41** [comment]
Thanks a lot @vestrel00 really appreciate your help. I created a [public repo](https://github.com/kdani41/Dagger2_11_support) for this scenario. If you get a chance please have a look :)

**Comment by kdani41** [comment]
@vestrel00 Thanks a lot for your help. I got some help and figured out where exactly I was screwing up. So the main problem occurred because I was not including [this](https://github.com/vestrel00/android-dagger-butterknife-mvp/blob/master/app/src/main/java/com/vestrel00/daggerbutterknifemvp/ui/example_1/Example1ActivityModule.java#L58) in my sub modules and I saw you have clearly explained in the comment why it is strongly required for all the child components. 

So, my bad and i'm closing this issue. Thanks again for sharing this solid example of using all this important libraries. I really appreciate your efforts 👍
